lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -31,14 +31,16 @@ class KubernetesResource
     TIMEOUT_OVERRIDE_ANNOTATION = "kubernetes-deploy.shopify.io/timeout-override"
 
     class << self
-      def build(namespace:, context:, definition:, logger:, statsd_tags:)
+      def build(namespace:, context:, definition:, logger:, statsd_tags:, crds: [])
         opts = { namespace: namespace, context: context, definition: definition, logger: logger,
                  statsd_tags: statsd_tags }
         if definition["kind"].blank?
           raise InvalidTemplateError.new("Template missing 'Kind'", content: definition.to_yaml)
         elsif KubernetesDeploy.const_defined?(definition["kind"])
           klass = KubernetesDeploy.const_get(definition["kind"])
           klass.new(**opts)
+        elsif crd = crds.find { |crd| crd.kind == definition["kind"] } # Dynamic CRs
```

Comment:
We could avoid changing this file by creating the CR classes with the right names dynamically. And then finding the parent CRD on the first sync. I can't tell if its a bad idea though. Its safe as is, but would require, that the CR classes never have class ivars

Reply:
Do you have an example of us using class ivars in any other resources, or a use-case where class-ivars might be worthwhile for custom resources?

I'm uneasy about dynamically creating classes, particularly since the implementation I've chosen is very much black-box and doesn't rely on the particulars of a given resource, only the contract it exposes via its metadata and status.